### PR TITLE
Remove immutable in teams 2 - convert rest of fields

### PR DESCRIPTION
**Pull request by buoyad**
No more immutable in teams store 🎉. 
cc @keybase/y2ksquad

### CODE REVIEW

File: shared/reducers/teams.tsx [hunk 1]
```diff
@@ -173,42 +157,59 @@ export default (
...
         return
-      case TeamsGen.deleteChannelInfo:
-        draftState.teamNameToChannelInfos = draftState.teamNameToChannelInfos.deleteIn([
-          action.payload.teamname,
-          action.payload.conversationIDKey,
-        ])
+      }
+      case TeamsGen.deleteChannelInfo: {
+        const {teamname, conversationIDKey} = action.payload
+        const oldChannelInfos =
```
Comment (chrisnojima) [inline shared/reducers/teams.tsx]: likely don't have to make and set it at all during this delete operation

File: shared/reducers/teams.tsx [hunk 2]
```diff
@@ -173,42 +160,66 @@ export default (
...
-      case TeamsGen.deleteChannelInfo:
-        draftState.teamNameToChannelInfos = draftState.teamNameToChannelInfos.deleteIn([
-          action.payload.teamname,
-          action.payload.conversationIDKey,
-        ])
+      }
+      case TeamsGen.deleteChannelInfo: {
+        const {teamname, conversationIDKey} = action.payload
+        const oldChannelInfos = draftState.teamNameToChannelInfos.get(teamname)
+        if (oldChannelInfos) {
```
Comment (buoyad) [inline shared/reducers/teams.tsx]: I think this is what you mean, lmk if not

**Comment by buoyad** [comment]
Thanks @chrisnojima, made changed PTAL

File: shared/actions/teams.tsx [hunk 3]
```diff
@@ -302,7 +302,7 @@ const addToTeam = async (_: TypedState, action: TeamsGen.AddToTeamPayload) => {
 
 const reAddToTeam = async (state: TypedState, action: TeamsGen.ReAddToTeamPayload) => {
   const {teamname, username} = action.payload
-  const id = state.teams.teamNameToID.get(teamname, '')
+  const id = state.teams.teamNameToID.get(teamname) || ''
```
Comment (chrisnojima) [inline shared/actions/teams.tsx]: `??`

File: shared/reducers/teams.tsx [hunk 4]
```diff
@@ -48,73 +48,60 @@ export default (
...
-          draftState.teamDetails.get(action.payload.teamID) ||
-          Constants.makeTeamDetails({teamname: action.payload.teamname})
-        details.members = new Map(
-          [...members.entries()].map(([username, memberInfo]) => [username, memberInfo.toObject()])
+        const details = mapGetEnsureValue(
+          draftState.teamDetails,
+          teamID,
+          Constants.makeTeamDetails({teamname})
         )
+        draftState.teamDetails.get(teamID) || Constants.makeTeamDetails({teamname})
```
Comment (chrisnojima) [inline shared/reducers/teams.tsx]: ???

File: shared/team-building/container.tsx [hunk 5]
```diff
@@ -155,8 +154,9 @@ const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
   const userResults: Array<Types.User> =
     teamBuildingState.searchResults.get(trim(ownProps.searchString))?.get(ownProps.selectedService) ?? []
 
-  const preExistingTeamMembers: I.Map<string, MemberInfo> =
-    (ownProps.teamname && state.teams.teamNameToMembers.get(ownProps.teamname)) || I.Map<string, MemberInfo>()
+  const preExistingTeamMembers: Map<string, MemberInfo> =
+    (ownProps.teamname && state.teams.teamNameToMembers.get(ownProps.teamname)) ||
```
Comment (chrisnojima) [inline shared/team-building/container.tsx]: `??`

Comment (buoyad) [reply shared/team-building/container.tsx]: That'd allow `preExistingTeamMembers === ''` (which TS caught, that's nice)

**Comment by buoyad** [comment]
@chrisnojima 🙇 made changes PTAL

**Pull request merged**